packages/engine-server/src/drivers/sqlite/tables/HierarchyTableUtils.ts

```diff
@@ -133,4 +133,33 @@ export class HierarchyTableUtils {
       }
     });
   }
+
+  /**
+   * Given a domain,
+   * recursively grab all descendants of the domain
+   * until we run out
+   *
+   * return all (parent, child) pairs
+   *
+   * more on recursive CTEs: https://www.sqlite.org/lang_with.html
+   */
+  public static getAllInDomain(
+    db: Database,
+    domainId: string
+  ): ResultAsync<HierarchyTableRow[], SqliteError> {
+    const sql = [
```

Comment:
anyone have a better solution to annotate multiline sql queries?

Reply:
imo this is a pretty good solution. I've started to adopt it in some local changes.